neo/Ledger/Blockchain.cs

```diff
@@ -583,7 +583,18 @@ private void Persist(Block block)
                             using (ApplicationEngine engine = new ApplicationEngine(TriggerType.Application, tx_invocation, snapshot.Clone(), tx_invocation.Gas))
                             {
                                 engine.LoadScript(tx_invocation.Script);
-                                engine.Execute();
+                                foreach (IDebuggerPlugin plugin in Plugin.DebuggerPlugins)
+                                {
+                                    if (plugin.DebuggerActive)
+                                    {
+                                        plugin.OnExecute(engine);
+                                        break;
+                                    }
+                                }
+                                if (engine.GasConsumed == Fixed8.Zero)
```

Comment:
Maybe this line should not be included. What is this verification being included?

Reply:
This is to fall through and run the engine through the normal execution path in the case that no debugger plugin intercepted and handled the execution.

Reply:
This if is outside the plugin scope.
Plugins should not change or add extra conditions outside the plugin loop.

Reply:
Agree with @vncoelho , looks weird for me too

Reply:
Should I instead be creating a second engine object with the same parameters and acting on that in the debugger, then return and execute the original engine normally? 

Will this still leave staged storage state changes/events on the second execution of the same script that need to be cleared somehow, or will the fact the debugged engine object is simply ignored at the end of the run enough?